cps/kni.c

```diff
@@ -996,6 +949,7 @@ init_kni(const char *kni_kmod_path, unsigned int num_kni)
 		if (errno == EEXIST) {
 			CPS_LOG(NOTICE, "%s: %s already inserted\n",
 				__func__, kni_kmod_path);
+			print_kmod_info(kni_kmod_path);
```

Comment:
`print_kmod_info()` is only called when the module is already loaded. In this condition, we already have the content of the module (i.e. variable `file`), so we can pass the module instead of its location. This would avoid another call to `grab_file()`, which is non-trivial operation and the second call may return another thing if the file was changed (I know, I'm paranoid with race conditions). Moreover, this change would not limit the interface of `print_kmod_info()` since a future caller could still call `grab_file()` before `print_kmod_info()`.